### PR TITLE
feat: regular board add remove edit move board columns

### PR DESCRIPTION
**Pull request by patricia-mdias**
Relates to #741

## Screenshots (if visual changes)
image

## Proposed Changes

  - Allow privileged users to edit columns names and remove or add up to 4 columns.

### CODE REVIEW

File: frontend/src/components/Board/Settings/index.tsx [hunk 1]
```diff
@@ -278,161 +317,201 @@ const BoardSettings = ({
       </Dialog.Header>
       <FormProvider {...methods}>
         <StyledForm
-          onSubmit={methods.handleSubmit(({ title, maxVotes }) => updateBoard(title, maxVotes))}
+          onSubmit={methods.handleSubmit(
+            ({ title, maxVotes, column1title, column2title, column3title, column4title }) => {
+              // eslint-disable-next-line @typescript-eslint/no-unused-vars
+              const updatedColumns = [column1title, column2title, column3title, column4title];
```
Comment (nunocaseiro) [inline frontend/src/components/Board/Settings/index.tsx:323]: Is this necessary or is it possible to delete?

File: frontend/src/components/Board/Settings/index.tsx [hunk 2]
```diff
@@ -278,161 +317,201 @@ const BoardSettings = ({
...
+                        <ConfigurationSwitchSettings
+                          handleCheckedChange={handleIsPublicChange}
+                          isChecked={switchesState.isPublic}
+                          text="If you make this board public anyone with the link to board can access it. Where to find the link? Just copy the URL of the board itself and share it."
+                          title="Make board public"
                         />
+                      )}
+                  </ConfigurationSettings>
 
+                  {isSubBoard && (isStakeholderOrAdmin || isOwner || isSAdmin || isResponsible) && (
```
Comment (nunocaseiro) [inline frontend/src/components/Board/Settings/index.tsx:395]:
```suggestion
                  {isSubBoard && (hasPermissions) && (
```

Can you the extract the roles verification to a variable `hasPermissions`? 
`const hasPermissions = isStakeholderOrAdmin || isOwner || isSAdmin || isResponsible`

File: frontend/src/components/Board/Settings/index.tsx [hunk 3]
```diff
@@ -278,161 +317,201 @@ const BoardSettings = ({
...
+                              responsible?.lastName ?? '-',
+                            )}
+                          />
+                        </Flex>
+                      </ConfigurationSwitchSettings>
+                    </TeamResponsibleSettings>
+                  )}
+                  {isRegularBoard && (
+                    <ColumnSettings>
+                      {/* '44vh' */}
```
Comment (nunocaseiro) [inline frontend/src/components/Board/Settings/index.tsx:485]:
```suggestion
```

File: frontend/src/components/Board/Settings/index.tsx [hunk 4]
```diff
@@ -278,161 +317,201 @@ const BoardSettings = ({
...
+                        {editColumns.map((title, index) => (
+                          <ColumnBoxAndDelete
+                            title={title}
+                            index={index}
+                            key={`column${index + 1}`}
+                            disableDeleteColumn={editColumns.length === 1}
+                            handleDeleteColumn={() => {
+                              const arrayWithoutColumn = editColumns.map((column) => column);
+                              arrayWithoutColumn.splice(index, 1);
+                              setEditColumns(arrayWithoutColumn);
```
Comment (nunocaseiro) [inline frontend/src/components/Board/Settings/index.tsx:496]: Can you move this code to a function?

File: frontend/src/components/Board/Settings/index.tsx [hunk 5]
```diff
@@ -278,161 +317,201 @@ const BoardSettings = ({
...
+                            }}
+                          />
+                        ))}
+                        {editColumns.length < 4 && (
+                          <AddColumnButton
+                            onAddColumn={() => {
+                              const arrayWithColumn = editColumns.map((column) => column);
+                              arrayWithColumn.push('');
+                              setEditColumns(arrayWithColumn);
```
Comment (nunocaseiro) [inline frontend/src/components/Board/Settings/index.tsx:505]: same here

**Review by nunocaseiro — CHANGES_REQUESTED** [review]

File: frontend/src/components/Board/Settings/index.tsx (hunk 1 above)
Comment (patricia-mdias) [reply frontend/src/components/Board/Settings/index.tsx:323]: I only had this line to use when I connect to the api. I can remove it.

**Review by nunocaseiro — APPROVED** [review]

**Pull request merged**